core/chains/evm/logpoller/orm.go

```diff
@@ -0,0 +1,117 @@
+package logpoller
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/sqlx"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/pg"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+type ORM struct {
+	chainID *big.Int
+	q       pg.Q
+}
+
+// NewORM creates an ORM scoped to chainID.
+func NewORM(chainID *big.Int, db *sqlx.DB, lggr logger.Logger, cfg pg.LogConfig) *ORM {
+	namedLogger := lggr.Named("ORM")
+	q := pg.NewQ(db, namedLogger, cfg)
+	return &ORM{
+		chainID: chainID,
+		q:       q,
+	}
+}
+
+// InsertBlock is idempotent to support replays.
+func (o *ORM) InsertBlock(h common.Hash, n int64, qopts ...pg.QOpt) error {
+	q := o.q.WithOpts(qopts...)
+	_, err := q.Exec(`INSERT INTO log_poller_blocks (evm_chain_id, block_hash, block_number, created_at) 
+      VALUES ($1, $2, $3, NOW()) ON CONFLICT DO NOTHING`, utils.NewBig(o.chainID), h[:], n)
+	return err
+}
+
+func (o *ORM) SelectBlockByHash(h common.Hash, qopts ...pg.QOpt) (*LogPollerBlock, error) {
+	q := o.q.WithOpts(qopts...)
+	var b LogPollerBlock
+	if err := q.Get(&b, `SELECT * FROM log_poller_blocks WHERE block_hash = $1 AND evm_chain_id = $2`, h, utils.NewBig(o.chainID)); err != nil {
+		return nil, err
+	}
+	return &b, nil
+}
+
+func (o *ORM) SelectBlockByNumber(n int64, qopts ...pg.QOpt) (*LogPollerBlock, error) {
+	q := o.q.WithOpts(qopts...)
+	var b LogPollerBlock
+	if err := q.Get(&b, `SELECT * FROM log_poller_blocks WHERE block_number = $1 AND evm_chain_id = $2`, n, utils.NewBig(o.chainID)); err != nil {
+		return nil, err
+	}
+	return &b, nil
+}
+
+func (o *ORM) SelectLatestBlock(qopts ...pg.QOpt) (*LogPollerBlock, error) {
+	q := o.q.WithOpts(qopts...)
+	var b LogPollerBlock
+	if err := q.Get(&b, `SELECT * FROM log_poller_blocks WHERE block_number = (SELECT MAX(block_number) FROM log_poller_blocks) AND evm_chain_id = $1`, utils.NewBig(o.chainID)); err != nil {
```

Comment:
Seems like this would fail if you have more than one chains? Assume chain A is on height 100, B on height 101, this would try to pick a block from A with height 101 which doesn't exist. Do we have tests for this scenario?

Also I think you could do away with the subquery completely?
```suggestion
	if err := q.Get(&b, `SELECT * FROM log_poller_blocks WHERE evm_chain_id = $1 ORDER BY block_number DESC LIMIT 1`, utils.NewBig(o.chainID)); err != nil {
```